sphinx/config.py

```diff
@@ -159,30 +148,27 @@ class Config(object):
                                 'env'),
     )  # type: Dict[unicode, Tuple]
 
-    def __init__(self, dirname, filename, overrides, tags):
-        # type: (unicode, unicode, Dict, Tags) -> None
+    def __init__(self, *args):
+        # type: (Any) -> None
+        if len(args) == 4:
+            # old style arguments: (dirname, filename, overrides, tags)
+            warnings.warn('The argument of Config() class has been changed. '
+                          'Use Config.read() to read configuration from conf.py.',
+                          RemovedInSphinx30Warning)
+            dirname, filename, overrides, tags = args
+            config = eval_config_file(dirname, filename, tags)
+        else:
+            # new style arguments: (config={}, overrides={})
```

Comment:
This allows to create a vanilla `Config` object easily.
I separated constructor which reading `conf.py` to `Config.read()` method.